contrib/database/sql/conn.go

```diff
@@ -103,16 +130,27 @@ func (tc *tracedConn) Ping(ctx context.Context) (err error) {
 	start := time.Now()
 	if pinger, ok := tc.Conn.(driver.Pinger); ok {
 		err = pinger.Ping(ctx)
+		tc.tryTrace(ctx, queryTypePing, "", start, err)
+		if err != nil {
+			return err
+		}
 	}
-	tc.tryTrace(ctx, queryTypePing, "", start, err)
+
 	return err
 }
 
 func (tc *tracedConn) QueryContext(ctx context.Context, query string, args []driver.NamedValue) (rows driver.Rows, err error) {
 	start := time.Now()
 	if queryerContext, ok := tc.Conn.(driver.QueryerContext); ok {
-		rows, err := queryerContext.QueryContext(ctx, query, args)
-		tc.tryTrace(ctx, queryTypeQuery, query, start, err)
+		sqlCommentCarrier := tracer.SQLCommentCarrier{}
```

Comment:
I feel like these 7 lines of code are repeated quite a bit.

Perhaps it might be nice to have a helper function that returns both the commented query and the "stop span" deferred function so it can be simplified to just 1 or 2 lines?